[Glitch] Refactor styles to use logical properties for positioning

Plastikmensch opened the pull request:
Port babd86e594bbdd6dcc512d522ccbb4b24c1355e3 to glitch-soc

~Sorry, I don't have the energy to double check this, so it needs a thorough review.~
Should be okay now

I couldn't find any documentation about `border-inset-inline-start` existing, so I replaced it with `border-inline-start`.
Kinda surprised stylelint doesn't flag unknown properties.

Reverted changes to `float:` as `inline-start` and `inline-end` are poorly supported by browsers.

Plastikmensch commented:
Hmh. This borks the reactions bar on announcements, but I'm not sure how to fix that since the inline style of `react-swipeable-view-container` always takes precedence.

ClearlyClaire commented:
Another thing I missed is that `float: inline-start` is poorly supported across browsers: https://github.com/mastodon/mastodon/pull/23944#issuecomment-1493408216

Plastikmensch commented:
> Another thing I missed is that `float: inline-start` is poorly supported across browsers: [mastodon#23944 (comment)](https://github.com/mastodon/mastodon/pull/23944#issuecomment-1493408216)

Reverted the changes to `float:` and restored some of the deleted rules in `rtl.scss` containing it.
Also noticed I forgot to break up 4 value margin properties, which I also did.

Plastikmensch commented:
Implemented announcement fix from 45848d654709678ba35d9e97e0b3df3e3916cd75
Removed some more now unneeded classes from `rtl.scss`
Fixed some missed properties.

The embed view is still somewhat borked in rtl, but I'm not sure if that should be fixed in a follow up.

ClearlyClaire commented on app/javascript/flavours/glitch/styles/rtl.scss:
```diff
@@ -1,131 +1,43 @@
-    margin-left: 10px;
-  }
-
-  .display-name {
-    text-align: right;
-  }
```
This causes a layout change because of the `bdi` tag.

That being said, I notice now this just swaps the behavior that currently exists with right-to-left names in a left-to-right interface, so addressing this is probably out of scope for this PR.

(The misalignment is something I failed to notice upstream but is there too, so out of scope for this PR as well)

## Before

![image](https://user-images.githubusercontent.com/384364/230865754-caaee89e-0312-4120-9465-e83eae148f3e.png)

## After

![image](https://user-images.githubusercontent.com/384364/230866040-fc78d89a-b87c-4d49-af65-60261c663f07.png)

ClearlyClaire commented:
> The embed view is still somewhat borked in rtl, but I'm not sure if that should be fixed in a follow up.

What exactly is messed up, and is that a regression? I haven't noticed any change

Plastikmensch commented:
> > The embed view is still somewhat borked in rtl, but I'm not sure if that should be fixed in a follow up.
> 
> What exactly is messed up, and is that a regression? I haven't noticed any change

Before:
![Screenshot_2023-04-10_12-23-22](https://user-images.githubusercontent.com/29483052/230884320-2e804f84-c5ac-492a-b0d5-f75d0275aadc.png)

After:
![Screenshot_2023-04-10_12-09-16](https://user-images.githubusercontent.com/29483052/230881682-a8212ff5-0d8c-472a-ab99-ef6bb5c30f06.png)

I think the order in the second screenshot is how it should be, but the margin applies to the wrong element.

ClearlyClaire commented:
Merged, but I could not reproduce the issues you pointed out wrt. embeds. How do you see these issues exactly?

Plastikmensch commented:
> Merged, but I could not reproduce the issues you pointed out wrt. embeds. How do you see these issues exactly?

That's odd.
I had issues with my local dev instance though, so I can't rule out that it's an issue on my end (though, I really have no idea what could've caused this)
I basically just clicked on a random toot and appended /embed to the url.
But I'm also now unable to reproduce it with this merged.
